contracts/attesters/hydra-s1/HydraS1AccountboundAttester.sol

```diff
@@ -0,0 +1,258 @@
+   * @param availableRootsRegistryAddress Registry storing the available groups for this attester (e.g roots of registry merkle trees)
+   * @param commitmentMapperAddress commitment mapper's public key registry
+   * @param collectionIdFirst Id of the first collection in which the attester is supposed to record
+   * @param collectionIdLast Id of the last collection in which the attester is supposed to record
+   */
+  constructor(
+    address attestationsRegistryAddress,
+    address hydraS1VerifierAddress,
+    address availableRootsRegistryAddress,
+    address commitmentMapperAddress,
+    uint256 collectionIdFirst,
+    uint256 collectionIdLast
+  )
+    HydraS1SimpleAttester(
+      attestationsRegistryAddress,
+      hydraS1VerifierAddress,
+      availableRootsRegistryAddress,
+      commitmentMapperAddress,
+      collectionIdFirst,
+      collectionIdLast
+    )
+  {}
+
+  /*******************************************************
+    MANDATORY FUNCTIONS TO OVERRIDE FROM ATTESTER.SOL
+  *******************************************************/
+
+  /**
+   * @dev Returns the actual attestations constructed from the user request
+   * @param request users request. Claim of having an account part of a group of accounts
+   * @param proofData snark public input as well as snark proof
+   */
+  function buildAttestations(Request calldata request, bytes calldata proofData)
+    public
+    view
+    virtual
+    override(IAttester, HydraS1SimpleAttester)
+    returns (Attestation[] memory)
+  {
+    Attestation[] memory attestations = super.buildAttestations(request, proofData);
+
+    uint256 nullifier = proofData._getNullifier();
+    attestations[0].extraData = abi.encodePacked(
+      attestations[0].extraData,
+      encodeNullifierAndNullifierBurnCount(nullifier, attestations[0].owner)
+    );
+
+    return (attestations);
+  }
+
+  /*******************************************************
+    OPTIONAL HOOK VIRTUAL FUNCTIONS FROM ATTESTER.SOL
+  *******************************************************/
+  /**
+   * @dev Hook run before recording the attestation.
+   * Throws if nullifier already used, not a renewal, and nullifier on cooldown.
+   * @param request users request. Claim of having an account part of a group of accounts
+   * @param proofData provided to back the request. snark input and snark proof
+   */
+  function _beforeRecordAttestations(Request calldata request, bytes calldata proofData)
+    internal
+    virtual
+    override
+  {
+    uint256 nullifier = proofData._getNullifier();
+    address nullifierDestination = _getDestinationOfNullifier(nullifier);
+
+    HydraS1Claim memory claim = request._claim();
+
+    // check if the nullifier has already been used previously, if so it may be on cooldown
+    if (nullifierDestination != address(0) && nullifierDestination != claim.destination) {
+      uint32 cooldownDuration = _getCooldownDurationForGroupId(claim.groupId);
```

Comment:
Use `claim.groupProperties.groupIndex` (internalCollectionId) instead of claim.groupId because the groupId is the hash(groupIndex,generationTimestamp,isScore) 
We should rename also the mapping _groupIdCooldowns to groupIndex or internalCollectionId instead of groupId

Reply:
+ Add the revert here if cooldownDuration == 0